async-gateway/service.go

```diff
@@ -81,14 +81,30 @@ func (s *service) GetWorkload(id string) (GetWorkloadResponse, error) {
 	log := s.logger.With(zap.String("id", id))
 
 	// download workload status
-	statusPath := fmt.Sprintf("%s/%s/status", prefix, id)
-	log.Debug("downloading status file", zap.String("path", statusPath))
-	statusBuf, err := s.storage.Download(statusPath)
+	statusPath := fmt.Sprintf("%s/%s/*", prefix, id)
+	log.Debug("checking status", zap.String("path", statusPath))
+	files, err := s.storage.List(fmt.Sprintf("%s/%s", prefix, id))
 	if err != nil {
 		return GetWorkloadResponse{}, err
 	}
 
-	status := Status(statusBuf[:])
+	status := StatusInQueue
+	for _, file := range files {
+		fileStatus := Status(file)
+
+		if !fileStatus.Valid() {
+			status = fileStatus
+			break
+		}
```

Comment:
I don't get this part. Shouldn't an error be returned if the status is not valid?

Reply:
I think I agree. The reason I let it pass was that it would have gotten caught in the switch case for the `default` case.